### PR TITLE
Feat changed "is_confirmed" to "confirmation_state"

### PR DESCRIPTION
**Pull request by tofarr**
**This parameter is not a boolean, so we should not name it as though it were.**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Simple find & replace: `is_confirmed` -> `confirmation_state`**

### CODE REVIEW

**Comment by ngduyanhece** [comment]
The pull request focuses on renaming the parameter `is_confirmed` to `confirmation_state` across multiple files to better reflect its non-boolean nature. This change is consistent and improves code readability by using a more descriptive name.

**Overall Feedback:**

The changes are straightforward and well-executed, ensuring that the new parameter name is consistently applied across the codebase. This improves the clarity of the code by using a name that accurately describes the parameter's purpose and possible values.

**Score:** 95/100

**Code Suggestions:**

1. **File:** `frontend/src/services/actions.ts`
   - **Language:** TypeScript
   - **Suggestion:** Consider using a switch statement instead of multiple if conditions for checking `confirmation_state` values. This can improve readability and maintainability if more states are added in the future.
   - **Relevant Line:** `+      !message.args.confirmation_state ||`

2. **File:** `openhands/runtime/runtime.py`
   - **Language:** Python
   - **Suggestion:** When checking for `confirmation_state`, consider using an enumeration for `ActionConfirmationStatus` to improve code readability and reduce the risk of typos in string comparisons.
   - **Relevant Line:** `+            and action.confirmation_state == ActionConfirmationStatus.AWAITING_CONFIRMATION`

3. **File:** `openhands/security/invariant/analyzer.py`
   - **Language:** Python
   - **Suggestion:** Ensure that all instances where `confirmation_state` is used are updated to handle any potential new states that might be introduced in the future.
   - **Relevant Line:** `+            and event.confirmation_state == 'awaiting_confirmation'`

The renaming of `is_confirmed` to `confirmation_state` is a positive change that enhances code clarity. Here are some suggestions:

1. In `frontend/src/services/actions.ts`, consider using a switch statement for `confirmation_state` checks to improve readability.
2. In `openhands/runtime/runtime.py`, use an enumeration for `ActionConfirmationStatus` to avoid string comparison errors.
3. In `openhands/security/invariant/analyzer.py`, ensure future-proofing by handling potential new states for `confirmation_state`.

File: frontend/src/services/actions.ts [hunk 1]
```diff
@@ -57,8 +57,8 @@ const messageActions = {
       store.dispatch(addAssistantMessage(message.args.thought));
     }
     if (
-      !message.args.is_confirmed ||
-      message.args.is_confirmed !== "rejected"
+      !message.args.confirmation_state ||
```
Comment (ngduyanhece) [inline frontend/src/services/actions.ts:60]: Use a switch statement for `confirmation_state` checks.

File: openhands/security/invariant/analyzer.py [hunk 2]
```diff
@@ -137,8 +137,8 @@ async def should_confirm(self, event: Event) -> bool:
         return (
             risk is not None
             and risk < self.settings.get('RISK_SEVERITY', ActionSecurityRisk.MEDIUM)
-            and hasattr(event, 'is_confirmed')
-            and event.is_confirmed == 'awaiting_confirmation'
+            and hasattr(event, 'confirmation_state')
+            and event.confirmation_state == 'awaiting_confirmation'
```
Comment (ngduyanhece) [inline openhands/security/invariant/analyzer.py:141]: Ensure future-proofing by handling potential new states.

Comment (tofarr) [reply openhands/security/invariant/analyzer.py]: Done

File: frontend/src/services/actions.ts (hunk 1 above)
Comment (tofarr) [reply frontend/src/services/actions.ts]: The presence check was redundant

**Review by rbren — APPROVED** [review]

**Pull request merged**